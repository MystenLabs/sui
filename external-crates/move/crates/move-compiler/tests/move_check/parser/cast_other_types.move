module a::m {
    fun t() {
        // we only allow numeric types
        (0 as ());
        (0 as &u64);
    }
}
