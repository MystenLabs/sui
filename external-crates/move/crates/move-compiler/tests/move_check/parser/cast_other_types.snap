error[E04003]: built-in operation not supported
  ┌─ tests/move_check/parser/cast_other_types.move:4:10
  │
4 │         (0 as ());
  │          ^    -- Found: '()'. But expected: 'u8', 'u16', 'u32', 'u64', 'u128', 'u256'
  │          │     
  │          Invalid argument to 'as'

error[E04003]: built-in operation not supported
  ┌─ tests/move_check/parser/cast_other_types.move:5:10
  │
5 │         (0 as &u64);
  │          ^    ---- Found: '&u64'. But expected: 'u8', 'u16', 'u32', 'u64', 'u128', 'u256'
  │          │     
  │          Invalid argument to 'as'

